### PR TITLE
backport: bitcoin#19200, #19405, #20282

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
Bitcoin backports with breaking changes

## What was done?
 - bitcoin/bitcoin#20282
 - bitcoin/bitcoin#19200
 - bitcoin/bitcoin#19405

## How Has This Been Tested?
Run unit/functional tests

### RPC:
- `upgradewallet` now returns object for future extensibility (#20282)

- `getnetworkinfo` now returns fields `connections_in`, `connections_out`,
  `connections_mn_in`, `connections_mn_out`, `connections_mn`
  that provide the number of inbound and outbound peer
  connections. These new fields are in addition to the existing `connections`
  field, which returns the total number of peer connections. Old fields
  `inboundconnections`, `outboundconnections`, `inboundmnconnections`,
  `outboundmnconnections` and `mnconnections` are removed (#19405)

- Backwards compatibility has been dropped for two `getaddressinfo` RPC
  deprecations, as notified in the 19.1.0 and 19.2.0 release notes.
  The deprecated `label` field has been removed as well as the deprecated `labels` behavior of
  returning a JSON object containing `name` and `purpose` key-value pairs. Since
  20.1, the `labels` field returns a JSON array of label names. (#19200)

### CLI

- The `connections` field of `bitcoin-cli -getinfo` is expanded to return a JSON
  object with `in`, `out` and `total` numbers of peer connections and `mn_in`,
  `mn_out` and `mn_total` numbers of verified mn connections. It previously
  returned a single integer value for the total number of peer connections. (#19405)

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
> backport: bitcoin#20282 breaking changes! #5823

but yet

> ## Breaking Changes
> N/A

**Comment by knst** [comment]
> but yet

that's a draft, I hope to bring there more backports before v21 will happen! updated anyway to new placeholder

File: doc/release-notes-19200.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Wallet
+
+- Backwards compatibility has been dropped for two `getaddressinfo` RPC
+  deprecations, as notified in the 19.1.0 and 19.2.0 release notes.
+  The deprecated `label` field has been removed as well as the deprecated `labels` behavior of
+  returning a JSON object containing `name` and `purpose` key-value pairs. Since
+  20.1, the `labels` field returns a JSON array of label names. (#19200)
```
Comment (UdjinM6) [inline doc/release-notes-19200.md:7]: 19200: should point to this PR (5823) if to any at all

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -3736,7 +3736,6 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
                 {RPCResult::Type::NUM, "sigsrequired", /* optional */ true, "The number of signatures required to spend multisig output (only if script is multisig)."},
                 {RPCResult::Type::STR_HEX, "pubkey", /* optional */ true, "The hex value of the raw public key, for single-key addresses."},
                 {RPCResult::Type::BOOL, "iscompressed", /* optional */ true, "If the pubkey is compressed."},
-                {RPCResult::Type::STR, "label", "DEPRECATED. The label associated with the address. Defaults to \"\". Replaced by the labels array below."},
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:3739]: 19200: missed changes for `"label name"` below

File: doc/release-notes-19405.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+## Updated RPCs
+
+- `getnetworkinfo` now returns fields `connections_in`, `connections_out`,
+  `connections_mn_in`, `connections_mn_out`, `connections_mn`
+  that provide the number of inbound and outbound peer
+  connections. These new fields are in addition to the existing `connections`
+  field, which returns the total number of peer connections. Old fields
+  `inboundconnections`, `outboundconnections`, `inboundmnconnections`,
+  `outboundmnconnections` and `mnconnections` are removed (#19405)
```
Comment (UdjinM6) [inline doc/release-notes-19405.md:9]: 19405: s/19405/5823/

File: doc/release-notes-19405.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+## CLI
+
+- The `connections` field of `bitcoin-cli -getinfo` is expanded to return a JSON
```
Comment (UdjinM6) [inline doc/release-notes-19405.md:13]: 19405: s/bitcoin/dash/

File: doc/release-notes-19405.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  object with `in`, `out` and `total` numbers of peer connections and `mn_in`,
+  `mn_out` and `mn_total` numbers of verified mn connections. It previously
+  returned a single integer value for the total number of peer connections. (#19405)
```
Comment (UdjinM6) [inline doc/release-notes-19405.md:16]: 19405: s/19405/5823/

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by PastaPastaPasta — DISMISSED** [review]

File: src/wallet/rpcwallet.cpp [hunk 6]
```diff
@@ -3736,15 +3736,13 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
...
                 {RPCResult::Type::NUM_TIME, "timestamp", /* optional */ true, "The creation time of the key, if available, expressed in " + UNIX_EPOCH_TIME + "."},
                 {RPCResult::Type::STR_HEX, "hdchainid", /* optional */ true, "The ID of the HD chain."},
                 {RPCResult::Type::STR, "hdkeypath", /* optional */ true, "The HD keypath, if the key is HD and available."},
                 {RPCResult::Type::STR_HEX, "hdseedid", /* optional */ true, "The Hash160 of the HD seed."},
                 {RPCResult::Type::STR_HEX, "hdmasterfingerprint", /* optional */ true, "The fingerprint of the master key."},
                 {RPCResult::Type::ARR, "labels", "An array of labels associated with the address. Currently limited to one label but returned as an array to keep the API stable if multiple labels are enabled in the future.",
                 {
-                    {RPCResult::Type::STR, "label name", "The label name. Defaults to \"\"."},
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:3747]: 19200: hmmm... this is not what I meant... check the original PR, should be
```suggestion
                    {RPCResult::Type::STR, "label name", "Label name (defaults to \"\")."},
```
and drop
```
                    {RPCResult::Type::OBJ, "", "json object of label data",
                    {
                        {RPCResult::Type::STR, "name", "The label name. Defaults to \"\"."},
                        {RPCResult::Type::STR, "purpose", "The Purpose of the associated address (send or receive)."},
                    }},
```
block below

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK 9c54cb16de0e78874cfcb83138811d91f9f20f1a

**Pull request merged**